agents/appmotion_motor/sat1_motors_driver.py

```diff
@@ -0,0 +1,825 @@
+                                print("Specified motor is invalid - not starting jogging.")
+                                continue
+
+                        # Set the jog parameters
+                        motor.write(b'JS%1.3f\r' % (velocity)) #JS = Jog Speed
+                        motor.write(b'JA%i\r' % (accel)) #JA = Jog Acceleration
+                        motor.write(b'JL%i\r' % (accel)) #JL = Jog Decel
+
+                        # Start rotation
+                        motor.write(b'CJ\r') #CJ = Commence Jogging
+                        motor.flushInput()
+
+
+        def stopRotation(self, motor=MOTOR2):
+                """Stops jogging specifically for the rotation of MOTOR2."""
+                mList = self.genMotorList(motor)
+                for motor in mList:
+                        if not motor:
+                                print("Specified motor is invalid.")
+                                continue
+                        motor.write(b'SJ\r') #SJ = Stop Jogging
+                        motor.flushInput()
+
+        def resetAlarms(self, motor=ALL):
+                """Resets alarm codes present. Only advised if you have checked what the alarm is first!
+
+                Parameters:
+                motor (int/motor name) -- desired motor (default ALL)
+                """
+
+                mList = self.genMotorList(motor)
+                for motor in mList:
+                        if not motor:
+                                print("Specified motor is invalid.")
+                                continue
+                        motor.write(b'AR\r')
+                        motor.flushInput()
+
+
+        def closeConnection(self, motor=ALL):
+                """
+                Close the connection to the serial controller for the specified motor.
+                motor = 1,2,3. Default is 3, or all motors.
+                """
+                mList = self.genMotorList(motor)
+                for motor in mList:
+                        if not motor:
+                                print("Specified motor is invalid - no connection to close.")
+                                continue
+                        motor.sock.close()
+                print("Connection to serial controller disconnected.")
+
+        def reconnectMotor(self, motor=ALL):
+                """
+                Reestablish connection with specified motor.
+                motor = 1,2,3. 3 is all motors. (Default 3)
+                """
+                mList = self.genMotorList(motor)
+                for mot in mList:
+                        if not mot:
+                                print("Specified motor is invalid - no connection to close.")
+                                continue   
+                        print(f"port: {mot.port}")
+                        mot.sock.connect(mot.port)
+                print(f"Connection with motor{motor} has been reestablished.")
+
+
+class PowerControl(object):
```

Comment:
Just curious, this isn't used anywhere, what's it for?

Reply:
That was from an abandoned plan to get programmable power strips and put them on our beam-mapper. We never ended up getting the power strips so I removed the code from the agent, and saved this code elsewhere